src/test/kotlin/uk/gov/dluhc/printapi/service/PrintResponseFileReadinessServiceTest.kt

```diff
@@ -0,0 +1,124 @@
+
+    companion object {
+        const val DIRECTORY = "/some-directory"
+    }
+
+    @Test
+    fun `should not mark print response file and not send to queue when no files are present`() {
+        // Given
+
+        given(sftpProperties.printResponseDownloadDirectory).willReturn(DIRECTORY)
+        given(sftpService.identifyFilesToBeProcessed(any())).willReturn(emptyList())
+
+        // When
+        printResponseFileReadinessService.markAndSubmitPrintResponseFileForProcessing()
+
+        // Then
+        verify(sftpService).identifyFilesToBeProcessed(DIRECTORY)
+        verifyNoInteractions(printMessagingService)
+        verifyNoMoreInteractions(sftpService)
+    }
+
+    @Test
+    fun `should mark print response file and send to queue when files are present`() {
+        // Given
+        val matchingFileName1 = "status-20221101171156056.json"
+        val matchingFileName2 = "status-20221101171156057.json"
+        val renamedFileName1 = "$matchingFileName1.processing"
+        val renamedFileName2 = "$matchingFileName2.processing"
+
+        given(sftpProperties.printResponseDownloadDirectory).willReturn(DIRECTORY)
+        given(sftpService.identifyFilesToBeProcessed(any())).willReturn(listOf(matchingFileName1, matchingFileName2))
+        given(sftpService.markFileForProcessing(eq(DIRECTORY), eq(matchingFileName1))).willReturn(renamedFileName1)
+        given(sftpService.markFileForProcessing(eq(DIRECTORY), eq(matchingFileName2))).willReturn(renamedFileName2)
+
+        // When
+        printResponseFileReadinessService.markAndSubmitPrintResponseFileForProcessing()
+
+        // Then
+        verify(sftpService).identifyFilesToBeProcessed(DIRECTORY)
+        verify(sftpService).markFileForProcessing(DIRECTORY, matchingFileName1)
+        verify(sftpService).markFileForProcessing(DIRECTORY, matchingFileName2)
+        verify(printMessagingService).submitPrintResponseFileForProcessing(
+            ProcessPrintResponseFileMessage(DIRECTORY, renamedFileName1)
+        )
+        verify(printMessagingService).submitPrintResponseFileForProcessing(
+            ProcessPrintResponseFileMessage(DIRECTORY, renamedFileName2)
+        )
+        verifyNoMoreInteractions(sftpService, printMessagingService)
+    }
+
+    @Test
+    fun `should continue marking print response file and sending to queue when some of files marking fails due to exception`() {
+        // Given
+        val matchingFile1ThatThrowsIOException = "status-20221101171156051.json"
+        val matchingFile3ThatThrowsIOException = "status-20221101171156053.json"
+        val matchingFileName2 = "status-20221101171156052.json"
+        val matchingFileName4 = "status-20221101171156054.json"
+        val renamedFileName2 = "$matchingFileName2.processing"
+        val renamedFileName4 = "$matchingFileName4.processing"
+        val expectedIOException = IOException("Unable to rename file")
```

Comment:
Much better; thank you 👍 Really nice and clear as to _why_ this test is relevant 👍